fud2/fud-core/src/run.rs

```diff
@@ -87,6 +87,77 @@ impl EmitBuild for EmitBuildFn {
     }
 }
 
+/// A config variable.
+#[derive(Debug, Clone)]
+pub enum ConfigVar {
+    /// The key for the config variable.
+    Var(String),
+    /// The key for the config variable followed by the value it should be if the key is not found.
+    VarOr(String, String),
+}
+
+/// The data required to emit a single op.
+pub struct OpEmitData {
```

Comment:
To be more specific, this is for a _particular, simple kind of op_ that is based on a bunch of data structures instead of a plain callback. (That is, a more generic system exists to define ops, and clients of this library do not _need_ to use this way.) Maybe renaming this struct to, like, `RuleOp` or something would help, and we could explain that this struct exists as a way to define ops that correspond 1:1 to Ninja rules?

Reply:
I don't really like the name `RuleOp` because it's perfectly fine for the struct map to to multiple rules (and more abstractly could get "out of sync" with the actual implementation). I do agree something better would be nice, but I'm more of mind to do something like `OrderedCommandOp`.

Reply:
Actually, I think I see what you are thinking about with `RuleOp`. Like these emits should correspond to structures which are emitted instead of the type of op.

I do still thing it would be good to pluralize `RuleOp` to `RulesOp` to emphasize it can emit multiple rules.

Reply:
Ah yes, definitely a good point! I like `RulesOp`.